### PR TITLE
Feature/13 questionnaire

### PR DESCRIPTION
**Pull request by f-lukas**
https://github.com/Pandora-IsoMemo/MapR/issues/13

### New Features
- user can fill out questionnaires and images are displayed depending on the answers

### CODE REVIEW

**Comment by f-lukas** [comment]
@arunge could you please review the changes? You can use the example zips and test the different behaviour for the old file and the new questionnaire file. Currently there is only one example image in the questionnaire file. This makes it difficult to "hit" the right values when filling out the questionnaire. So it would be good to have a quick talk before you try it.

I also changed the code structure quite a lot. I can tell you more about it but the "main" code is still included in the mappanel module. But it is now much cleaner because I outsourced some code into functions. However the new questionnaire functionality made the app more complex. I can give you a short introduction into the code before you start, if you want.

Thanks!

File: R/02-observeShowPlot.R [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+prepareQuestionnaireImage <- function(input, questionnaire) {
+  # Iterate over questions to get ids of question inputs
+  inputs <- lapply(questionnaire()$Questions, function(question) {
+    input[[paste0("question_", question$Question_ID)]]
+  })
+  imageInfos <- lapply(questionnaire()$Plots, function(plot) {
+    # Check if length of questions (inputs) is equal to length of required answers
+    if (length(plot$Answers) != length(inputs)) {
+      DataTools::tryCatchWithWarningsAndErrors(
+        expr = stop(""),
```
Comment (arunge) [inline R/02-observeShowPlot.R]: Interesting how you use `DataTools::tryCatchWithWarningsAndErrors()` in this situation, where you always expect an error if the condition is true.
One could also just use:
```
shinyjs::alert("Number of questions and answers in questionnaire does not match")
```
Or is there another reason why you use the function from DataTools?

**Comment by f-lukas** [comment]
@arunge I added a field `Fill_Value` for every question in the json object. I think this is more flexibel if Ricardo wants to provide pre filled questionnaires later. But this field can be still be "" if the inputs should not be filled. In the example_questionnaire.zipm, I specified the values that guarantee that a plot is displayed.

**Review by arunge — COMMENTED** [review]
I checked your code: very well structured! I will next pull and check your updates that we just discussed.

**Review by arunge — APPROVED** [review]
approved as discussed :blush:

**Pull request merged**